lib/sorbet-rails/typed_params.rb

```diff
@@ -0,0 +1,43 @@
+# typed: true
+require 'sorbet-coerce'
+
+module TypedParams
+  @extract_with_hard_errors = Proc.new do |&blk|
+    prev_inline_type_error_handler = T::Configuration.instance_variable_get(
+      :@inline_type_error_handler,
+    )
+    prev_call_validation_error_handler = T::Configuration.instance_variable_get(
+      :@call_validation_error_handler,
+    )
+    T::Configuration.inline_type_error_handler = nil
+    T::Configuration.call_validation_error_handler = nil
+    blk.call
+  ensure
+    T::Configuration.inline_type_error_handler = prev_inline_type_error_handler
+    T::Configuration.call_validation_error_handler = prev_call_validation_error_handler
+  end
+
+  def self.extract_params_with_hard_errors=(handler)
+    # For supporting customized sorbet error handlers
+    @extract_with_hard_errors = handler
+  end
+
+  def self.extract_params_with_hard_errors
+    @extract_with_hard_errors
+  end
+
+  # A sorbet-coerce wrapper for coercing typed controller parameters
+  define_singleton_method(:[]) do |type|
+    Class.new do
+      define_method(:extract!) do |params|
+        begin
+          TypedParams.extract_params_with_hard_errors.call {
+            T::Coerce[type].new.from(params.permit!.to_h)
+          }
+        rescue StandardError => e
```

Comment:
StandardError only? How about TypeError?

Reply:
I think TypeError is captured here too.